Tests/ApolloTests/StarWars/API.swift

```diff
@@ -323,6 +384,15 @@ public final class HeroDetailsQuery: GraphQLQuery {
     "  }" +
     "}"
 
+  public static let selectionSet: [Selection] = [
+    Field("hero", arguments: ["episode": Variable("episode")], type: .object(Data.Hero.self), selectionSet: [
+      Field("__typename", type: .nonNull(.scalar(String.self))),
+      Field("name", type: .nonNull(.scalar(String.self))),
+      FragmentSpread(Data.Hero.AsHuman.self),
```

Comment:
Inline fragments without a type condition (or with a type condition that always matches) currently get merged into the parent selection set by `apollo-codegen`. I don't think they would help avoid cache misses, because if `c` and `d` are unavailable, reading the selection set as a whole would still fail. In those cases, a named fragment with just `a` and `b` might make more sense.